src/DetectChanges/BCBreak/FunctionBased/ParameterNameChanged.php

```diff
@@ -0,0 +1,119 @@
+
+    private ReflectionFunctionAbstractName $formatFunction;
+
+    public function __construct()
+    {
+        $this->formatFunction = new ReflectionFunctionAbstractName();
+    }
+
+    public function __invoke(ReflectionFunctionAbstract $fromFunction, ReflectionFunctionAbstract $toFunction): Changes
+    {
+        $fromHadNoNamedArgumentsAnnotation = $this->methodHasNoNamedArgumentsAnnotation($fromFunction);
+        $toHasNoNamedArgumentsAnnotation   = $this->methodHasNoNamedArgumentsAnnotation($toFunction);
+
+        if ($fromHadNoNamedArgumentsAnnotation && ! $toHasNoNamedArgumentsAnnotation) {
+            return Changes::fromList(
+                Change::removed(
+                    sprintf(
+                        'The %s annotation was removed from %s',
+                        self::NO_NAMED_ARGUMENTS_ANNOTATION,
+                        $this->formatFunction->__invoke($fromFunction),
+                    ),
+                    true
+                )
+            );
+        }
+
+        if (! $fromHadNoNamedArgumentsAnnotation && $toHasNoNamedArgumentsAnnotation) {
+            return Changes::fromList(
+                Change::added(
+                    sprintf(
+                        'The %s annotation was added from %s',
+                        self::NO_NAMED_ARGUMENTS_ANNOTATION,
+                        $this->formatFunction->__invoke($fromFunction),
+                    ),
+                    true
+                )
+            );
+        }
+
+        if ($toHasNoNamedArgumentsAnnotation) {
+            return Changes::empty();
+        }
+
+        return Changes::fromIterator($this->checkSymbols(
+            $fromFunction->getParameters(),
+            $toFunction->getParameters()
+        ));
+    }
+
+    /**
+     * @param ReflectionParameter[] $from
+     * @param ReflectionParameter[] $to
+     *
+     * @return iterable|Change[]
+     */
+    private function checkSymbols(array $from, array $to): iterable
+    {
+        foreach (array_intersect_key($from, $to) as $index => $commonParameter) {
+            yield from $this->compareParameter($commonParameter, $to[$index]);
+        }
+    }
+
+    /**
+     * @return iterable|Change[]
+     */
+    private function compareParameter(ReflectionParameter $fromParameter, ReflectionParameter $toParameter): iterable
+    {
+        $fromName = $fromParameter->getName();
+        $toName   = $toParameter->getName();
+
+        if ($fromName === $toName) {
+            return;
+        }
+
+        yield Change::changed(
+            sprintf(
+                'Parameter %d of %s changed name from %s to %s',
+                $fromParameter->getPosition(),
+                $this->formatFunction->__invoke($fromParameter->getDeclaringFunction()),
+                $fromName,
+                $toName
+            ),
+            true
+        );
+    }
+
+    private function methodHasNoNamedArgumentsAnnotation(ReflectionFunctionAbstract $function): bool
```

Comment:
I wonder if any static analysis library will add some kind of PHP attribute like `#[NoNamedAttributes]` or something, we could look for that too, although I think we'd need to add support upstream in Better Reflection for attributes first, so that could be an improvement to come later.

Reply:
Possibly to be done once the ecosystem agrees on these annotations - too early for us to jump in on this